src/datawave_cli/dictionary_interactions.py

```diff
@@ -23,38 +22,46 @@ def __init__(self, args, log: logging.Logger = logging.getLogger('dictionary_int
     def get_pod_ip(self):
         return pods.get_specific_pod(pods.web_dictionary_info, self.namespace).pod_ip
 
-    def get_entire_dictionary(self, auths: str, data_types: str, file: str):
-        """
-        Display or save the entire dictionary of datawave, all data types.
+    def get_dictionary(self, auths: str, data_types: str, filename: str):
+        """Display or save the dictionary of datawave for the provided data types.
 
         Parameters
         ----------
         auths: str
-            comma delimited string containing all the auths to pull the dictionary for
+            comma delimited string containing all the auths to pull the dictionary with
         data_types: str
             comma delimited string comtaining the data types to pull the dictionary for
         file: str
             the name of where to save output
+
+        Raises
+        ------
+        RuntimeError:
+            Raised if an invalid response is recieved from the request
         """
         self.log.info("Getting the entire field dictionary in DataWave...")
         request = f"{self.base_url}/dictionary/data/v1/"
         data = {'auths': auths,
                 'dataTypeFilters': data_types}
         resp = requests.get(request, data=data, cert=self.cert, headers=self.headers, verify=False)
         log_http_response(resp, self.log)
-        if resp.status_code == 200:
+        fields = {}
+        try:
+            resp.raise_for_status()
             fields = self.parse_response(resp)
-            if file is None:
-                self.display_dictionary(fields, self.log)
+            if filename is None:
+                self.output_dictionary(self.log.info, fields)
             else:
-                self.save_dictionary(file, fields, self.log)
-            return fields
-        else:
-            return {}
+                with open(filename, 'a') as file:
```

Comment:
@DL-USAF Did we want this to be write instead of append?

Reply:
I think write would make more sense. Why do you want to keep old? but I'm okay with either.

Reply:
move the write logic to be in main, or a in a function that gets the dictionary info, then displays it.

Reply:
Done